internal/bundlereader/loaddirectory_test.go

```diff
@@ -29,7 +29,29 @@ func TestGetContent(t *testing.T) {
 		name               string
 		directoryStructure fsNode
 		expectedFiles      map[string][]byte
+		source             string
+		auth               bundlereader.Auth
+		expectedErr        string
 	}{
+		{
+			name: "ensure panic doesn't occur when InsecureSkipVerify is set to false (#3782)",
+			directoryStructure: fsNode{
+				name: "fleet.yaml",
+				contents: `namespace: fleet-helm-oci-with-auth-example
+		 helm:
+		   chart: "oci://ghcr.io/fleetqa/fleet-qa-examples/fleet-test-configmap-chart"
+		   version: "0.1.0"
+		   values:
+		     replicas: 2`,
+			},
+			source: "oci://ghcr.io/fleetqa/fleet-qa-examples/fleet-test-configmap-chart",
```

Comment:
This may not need actual OCI registry URLs; we could use dummy ones (e.g. `oci://foo/bar/baz`) to clarify that this fits in a unit test case and that exposing/fixing the issue does not involve actually reaching the registry.

Reply:
I do get inconsistent results. On my machine I could check for "no such host", on [Github](https://github.com/rancher/fleet/actions/runs/15607929114/job/43966719236) I get

```
2025/06/12 11:41:14 http: TLS handshake error from 127.0.0.1:35048: remote error: tls: bad certificate
```